File: solutions/go/isogram/1/isogram.go
package isogram

import "strings"

func IsIsogram(word string) bool {
	found := make(map[rune]bool)
    for _, rune := range strings.ToLower(word) {
        // only check alphabetical letters
        if rune < 'a' || rune > 'z' {
            continue
        }
        if found[rune] {
            return false
        }
        found[rune] = true
    }
    return true
}

File: solutions/javascript/isogram/1/isogram.js

export function isIsogram(word) {
  
  // Retrieve the letters from the word and count them (case insensitive).
  // If any count is greater than one, then it is not an isogram.
  let letters = word.toLowerCase().match(/[a-z]/g);
  if (!letters) return true;
  
  const frequencies = new Map();
  for (const letter of letters) {
    frequencies.set(letter, 1 + (frequencies.get(letter) || 0));
  }
  
  for (const count of frequencies.values()) {
    if (count > 1) return false;
  }
  return true;
};
